src/main/java/org/folio/marccat/integration/tools/TenantLoading.java

```diff
@@ -154,54 +161,77 @@ public void perform(TenantAttributes ta, Map<String, String> headers) {
       logger.debug("TenantLoading.perform No X-Okapi-Url header");
       return;
     }
-    Iterator<LoadingEntry> it = loadingEntries.iterator();
-    performR(okapiUrl, ta, headers, it);
+    performR(okapiUrl, headers, loadBibliographicSample);
   }
 
   /**
    * Perform R.
    *
-   * @param okapiUrl the okapi url
-   * @param ta the ta
-   * @param headers the headers
-   * @param it the it
+   * @param okapiUrl                the okapi url
+   * @param headers                 the headers
+   * @param it                      the it
+   * @param loadBibliographicSample
    */
-  private void performR(String okapiUrl, TenantAttributes ta,
-                        Map<String, String> headers, Iterator<LoadingEntry> it) {
+  private void performR(String okapiUrl, Map<String, String> headers, boolean loadBibliographicSample) {
 
-    loadData(okapiUrl, headers);
+    if (loadBibliographicSample) {
+      loadDataBibliographic(okapiUrl, headers);
+    }
+    loadDataAuthority(okapiUrl, headers);
   }
+
   /**
    * Load data.
    *
    * @param okapiUrl the okapi url
-   * @param headers the headers
-    */
-  private void loadData(String okapiUrl, Map <String, String> headers) {
+   * @param headers  the headers
+   */
+  private void loadDataBibliographic(String okapiUrl, Map<String, String> headers) {
     final String endPointUrl = okapiUrl + "/" + "marccat/load-from-file";
     logger.debug("Load data URL " + endPointUrl);
     HttpHeaders httpHeaders = new HttpHeaders();
     httpHeaders.add(Global.OKAPI_TENANT_HEADER_NAME, headers.get(Global.OKAPI_TENANT_HEADER_NAME));
     httpHeaders.add(Global.OKAPI_URL, headers.get(Global.OKAPI_URL));
```

Comment:
It looks like we are being selective with the headers we are passing on to Okapi. We should probably pass all the "X-Okapi-*" headers. See: https://github.com/folio-org/okapi/blob/master/doc/guide.md#x-okapi-headers

Reply:
The value of the constants of Global are "X-Okapi-Url" and "X-Okapi-Tenant"

Reply:
Yes, I understand, but those are the bare minimum headers. Note in the link I added in the above comment that there are many more X-Okapi-* headers that can be set. These other headers should also be passed to the client as well. Here is the salient bit from that link:

> If the module wishes to make a request to some other module, it should address it to the base URL from X-Okapi-Url, combined with the path it needs to access. It should pass along at least the X-Okapi-Token, and preferably also the X-Okapi-Request-Id. **In many cases it is easier to pass all X-Okapi headers along** - Okapi will drop those that could confuse things.

Reply:
This call goes through Okapi, we saw that he didn't need the others. We will add them all even if they are not used by Okapi. This change will also be made in the next PullRequest.